### ISSUE TITLE
Possible regression in F.interpolate on nightly build

### ISSUE THREAD

**Issue by aounleonardo** [pytorch/pytorch#54590]
## 🐛 Bug

The BatchNorm2d layer is providing different results on stable vs nightly build which is causing the rest of the face landmarks detection to be completely wrong.

## To Reproduce

1. Clone https://github.com/aounleonardo/face_alignment/
1. pip3 install numpy
1. pip3 install opencv-python
1. Build pytorch using `conda install pytorch torchvision cudatoolkit=11.1 -c pytorch -c conda-forge` and it should work
1. If you build with `conda install pytorch torchvision cudatoolkit=11.1 -c pytorch-nightly -c conda-forge` instead, it will fail and the results will be clear

I have identified the first line where a difference in the tensors appear, and it's this one: https://github.com/ibug-group/face_alignment/blob/master/ibug/face_alignment/fan/fan.py#L30

![image](https://user-images.githubusercontent.com/11409469/112321744-39613b80-8ca8-11eb-80e9-00c27199f3e2.png)

## Expected behavior
![image](https://user-images.githubusercontent.com/11409469/112321686-29e1f280-8ca8-11eb-8c95-67cb172c8c01.png)

## Environment
Collecting environment information...
PyTorch version: 1.9.0.dev20210319
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Enterprise
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce RTX 2070 Super with Max-Q Design
Nvidia driver version: 461.92
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.1

[pip3] torch==1.9.0.dev20210319

[pip3] torchvision==0.9.0.dev20210319
[conda] blas                      2.108                       mkl    conda-forge

[conda] blas-devel                3.9.0                     8_mkl    conda-forge

[conda] cudatoolkit               11.1.1               heb2d755_7    conda-forge

[conda] libblas                   3.9.0                     8_mkl    conda-forge

[conda] libcblas                  3.9.0                     8_mkl    conda-forge

[conda] liblapack                 3.9.0                     8_mkl    conda-forge

[conda] liblapacke                3.9.0                     8_mkl    conda-forge

[conda] mkl                       2020.4             hb70f87d_311    conda-forge

[conda] mkl-devel                 2020.4             h57928b3_312    conda-forge

[conda] mkl-include               2020.4             hb70f87d_311    conda-forge

[conda] numpy                     1.20.1           py38h0cc643e_0    conda-forge

[conda] pytorch                   1.9.0.dev20210319 py3.8_cuda11.1_cudnn8_0    pytorch-nightly

[conda] torchvision               0.9.0.dev20210319      py38_cu111    pytorch-nightly

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @csarofeen @ptrblck @xwang233

**Comment by aounleonardo** [comment]
I updated the repo with the test plan included in the readme https://github.com/aounleonardo/face_alignment/blob/nightly_build_issue/README.md

**Comment by aounleonardo** [comment]
I'm only printing [0,0,0] of each tensor in these examples, but I verified with torch.equal that before they enter the first block, they are indeed the same.

**Comment by ngimel** [comment]
Can you please post output of 
```
print(*torch.__config__.show().split("\n"), sep="\n")
```
for your nightly and 1.8 versions?

**Comment by ziky90** [comment]
I can add one more finding related to this issue. BatchNorm2d seems to work well in the PyTorch nightly on CPU, this issue happens only when the inference is being run on the GPU.

**Comment by aounleonardo** [comment]
@ngimel Thanks for your reply, I pushed the config files into batch_norm_test_resources/*.torchconfig, but here they are too:

1.8.0:
PyTorch built with:
  - C++ Version: 199711
  - MSVC 192829337
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 2019
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.4
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=C:/w/b/windows/tmp_bin/sccache-cl.exe, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL -openmp:experimental -DNDEBUG -DUSE_FBGEMM -DUSE_XNNPACK, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON, 

1.9.0:
PyTorch built with:
  - C++ Version: 199711
  - MSVC 192829337
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.8.0 (Git Hash 2c8d20640d5068e2d85e378b266644fe86220e84)
  - OpenMP 2019
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.4
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=C:/cb/pytorch_1000000000000/work/tmp_bin/sccache-cl.exe, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL -openmp:experimental -IC:/cb/pytorch_1000000000000/work/mkl/include -DNDEBUG -DUSE_FBGEMM -DUSE_XNNPACK, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.9.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON,

**Comment by aounleonardo** [comment]
@ngimel I debugged further, and it seems that the F.interpolate layer at line https://github.com/aounleonardo/face_alignment/blob/nightly_build_issue/ibug/face_alignment/fan/fan.py#L127 is the first one that creates this inconsistency.
I added snapshot tests around this line to prove this. First run with 1.8 then run again with the nightly build.

**Comment by ngimel** [comment]
Thanks, this is very helpful! cc @xwang233

**Comment by ngimel** [comment]
cc @bdhirsh. We've identified a likely cause and are working on a fix, thanks again for narrowing it down @aounleonardo!
